CallAutomation_OutboundCalling/CallAutomation_OutboundCalling/Program.cs

```diff
@@ -0,0 +1,108 @@
+});
+
+app.MapPost("/api/callbacks", async (CloudEvent[] cloudEvents, ILogger<Program> logger) =>
+{
+    foreach (var cloudEvent in cloudEvents)
+    {
+        CallAutomationEventBase @event = CallAutomationEventParser.Parse(cloudEvent);
+        logger.LogInformation($"Event Recieved. CallConnectionId[{@event?.CallConnectionId}], Type Name[{@event?.GetType().Name}]");
+        var callConnection = callAutomationClient.GetCallConnection(@event.CallConnectionId);
+        var callMedia = callConnection.GetCallMedia();
+
+        if (@event is CallConnected)
+        {
+            logger.LogInformation($"Start Recording...");
+            CallLocator callLocator = new ServerCallLocator(@event.ServerCallId);
+            _ = await callAutomationClient.GetCallRecording().StartAsync(new StartRecordingOptions(callLocator));
+
+            // prepare recognize tones
+            CallMediaRecognizeDtmfOptions callMediaRecognizeDtmfOptions = new CallMediaRecognizeDtmfOptions(new PhoneNumberIdentifier(targetPhonenumber), maxTonesToCollect: 1);
+            callMediaRecognizeDtmfOptions.Prompt = new FileSource(new Uri(callbackUriHost + "/audio/MainMenu.wav"));
+            callMediaRecognizeDtmfOptions.InterruptPrompt = true;
+            callMediaRecognizeDtmfOptions.InitialSilenceTimeout = TimeSpan.FromSeconds(5);
+            callMediaRecognizeDtmfOptions.InterToneTimeout = TimeSpan.FromSeconds(10);
+            callMediaRecognizeDtmfOptions.StopTones = new List<DtmfTone> { DtmfTone.Pound, DtmfTone.Asterisk };
+
+            // Send request to recognize tones
+            await callMedia.StartRecognizingAsync(callMediaRecognizeDtmfOptions);
+        }
+        if (@event is RecognizeCompleted)
+        {
+            // Play audio once recognition is completed sucessfully
+            logger.LogInformation($"RecognizeCompleted event received for call connection id: {@event.CallConnectionId}");
+            var recognizeCompleted = (RecognizeCompleted)@event;
+
+            string selectedTone = ((DtmfResult)recognizeCompleted.RecognizeResult).ConvertToString();
+
+            switch (selectedTone)
+            {
+                case "1":
+                    await callMedia.PlayToAllAsync(new FileSource(new Uri(callbackUriHost + "/audio/Confirmed.wav")));
+                    break;
+
+                case "2":
+                    await callMedia.PlayToAllAsync(new FileSource(new Uri(callbackUriHost + "/audio/Goodbye.wav")));         
+                    break;
+
+                default:
+                    await callConnection.HangUpAsync(true);
+                    break;
+            }
+        }
+        if ((@event is RecognizeFailed) || (@event is PlayCompleted) || (@event is PlayFailed))
+        {
+            logger.LogInformation($"Terminating Call. Event received for call connection id: {@event.CallConnectionId}");
```

Comment:
Could remove " Event received for call connection id: {@event.CallConnectionId}"